pandas/tests/indexes/numeric/test_numeric.py

```diff
@@ -312,7 +312,7 @@ def test_cant_or_shouldnt_cast(self, dtype):
 
     def test_view_index(self, simple_index):
         index = simple_index
-        msg = "Cannot change data-type for object array"
+        msg = "Cannot change data-type for array of references."
```

Comment:
We'll need to support both error messages probably as the error messages only changed on the dev build

Reply:
thanks, I added the previous error message to the new one.